pkg/extension/workspace/example/runtime/configdir/configdir_test.go

```diff
@@ -19,173 +19,184 @@ import (
+
+			// verify contents of the config dir
+			for _, fileName := range ctx.Opts.Example.GetFiles().GetNames() {
+				expected, err := ioutil.ReadFile(filepath.Join(test.workspaceDir, "expected/getenvoy_extension_run", fileName))
+				require.NoError(t, err)
+				actual, err := ioutil.ReadFile(filepath.Join(configDir.GetDir(), fileName))
+				require.NoError(t, err)
+				if test.isEnvoyTemplate(fileName) {
+					require.YAMLEq(t, string(expected), string(actual))
+				} else {
+					require.Equal(t, string(expected), string(actual))
 				}
-			},
-			Entry("envoy.tmpl.yaml", testCase{
-				workspaceDir: "testdata/workspace1",
-				isEnvoyTemplate: func(name string) bool {
-					return name == "envoy.tmpl.yaml" //nolint:goconst
-				},
-				expectBootstrap: expectValidBootstrap,
-			}),
-			Entry("envoy.tmpl.yaml + lds.tmpl.yaml + cds.yaml", testCase{
-				workspaceDir: "testdata/workspace2",
-				isEnvoyTemplate: func(name string) bool {
-					return name == "envoy.tmpl.yaml" || name == "lds.tmpl.yaml" || name == "cds.yaml"
-				},
-				expectBootstrap: expectValidBootstrap,
-			}),
-			Entry("envoy.tmpl.yaml: not a valid YAML", testCase{
-				workspaceDir: "testdata/workspace3",
-				isEnvoyTemplate: func(name string) bool {
-					return false
-				},
-				expectBootstrap: expectInvalidBootstrap,
-			}),
-			Entry("envoy.tmpl.yaml: invalid paths to `lds` and `cds` files", testCase{
-				workspaceDir: "testdata/workspace4",
-				isEnvoyTemplate: func(name string) bool {
-					return name == "envoy.tmpl.yaml"
-				},
-				expectBootstrap: expectValidBootstrap,
-			}),
-			Entry("envoy.tmpl.yaml: .txt configuration", testCase{
-				workspaceDir: "testdata/workspace8",
-				isEnvoyTemplate: func(name string) bool {
-					return name == "envoy.tmpl.yaml"
-				},
-				expectBootstrap: expectValidBootstrap,
-			}),
-		)
-	})
-
-	Describe("in case of invalid input", func() {
-		abs := func(path string) string {
-			path, err := filepath.Abs(path)
-			if err != nil {
-				panic(err)
 			}
-			return path
+		})
+	}
+}
+
+func runContext(workspace model.Workspace, example model.Example) *runtime.RunContext {
+	_, f := example.GetExtensionConfig()
+	return &runtime.RunContext{
+		Opts: runtime.RunOpts{
+			Workspace: workspace,
+			Example: runtime.ExampleOpts{
+				Name:    "default",
+				Example: example,
+			},
+			Extension: runtime.ExtensionOpts{
+				WasmFile: `/path/to/extension.wasm`,
+				Config:   *f,
+			},
+		},
+	}
+}
+
+func expectValidBootstrap(t *testing.T, bootstrap *envoybootstrap.Bootstrap) {
+	require.NotNil(t, bootstrap)
+	require.Equal(t, "/dev/null", bootstrap.GetAdmin().GetAccessLogPath())
+	require.Equal(t, "127.0.0.1", bootstrap.GetAdmin().GetAddress().GetSocketAddress().GetAddress())
+	require.Equal(t, uint32(9901), bootstrap.GetAdmin().GetAddress().GetSocketAddress().GetPortValue())
+}
+
+func expectInvalidBootstrap(t *testing.T, bootstrap *envoybootstrap.Bootstrap) {
+	require.Nil(t, bootstrap)
+}
```

Comment:
Can we just inline this since it seems just aliasing `require.Nil`?

Reply:
sgtm!

Reply:
actually it is messier due to `, msgAndArgs ...interface{}` which isn't entirely visible here. leaving as-is, but thanks for the suggestion as was worthwhile looking